### ISSUE TITLE
Maximum execution time exceeded on every page save

### ISSUE THREAD

**Issue by drnasin** [trilbymedia/grav-plugin-tntsearch#19]
I have cca 70 pages indexed. Every time I save the page the content will be saved but the page will hang and cause Execution time out

Grav reports problem here: tntsearch\vendor\teamtnt\tntsearch\src\Indexer\TNTIndexer.php, function saveDoclist, around here:

```
            try {
                $stmt->execute();
            } catch (\Exception $e) {
                //we have a duplicate
                echo $e->getMessage();
            }
```

Not sure what to do except for disabling the plugin...
UPDATE: once disabled, the whole process of saving a page became much faster...

**Comment by rhukster** [comment]
I've added options to let you disable page events, and just use the manual indexer process.  However, this is not a long-term solution.  The problem is I can't replicate this issue.  I need a site that has this problem that I could perhaps install locally and see what's going on.

**Comment by gamahachaa** [comment]
After upgrading to 2.0.4 while indexing, I stumbled upon 
`PHP Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 130968 bytes) in /home/qook/app/qook/system/src/Grav/Common/Page/Page.php on line 254`
Increasing to the max didn't help
changing the config filter to :
```yaml
filter:
  items:
    taxonomy@:
      category:
          - doc
```
Instead of 
```yaml
filter:
  items:
    - root@.descendants
```
did help.

**Comment by rhukster** [comment]
Can you please try the latest versus on in github develop branch?

**Comment by gamahachaa** [comment]
Tried. and the error is now triggered @the start of the indexing. the indexing then traces it is adding pages then stop after ~65 pages (same behavior - same page- as with master branch, except for the error that was triggered at the end).

I tried back with taxo filter. indexing went ok though "libpng warning: Interlace handling should be turned on when using png_read_image" is outputed several time before indexing starts adding pages (~700)

**Comment by rhukster** [comment]
Is it possible to get a dump of your site?

**Comment by gamahachaa** [comment]
I will try to narrow it down and get back to you (by MP?)

**Comment by gamahachaa** [comment]
I nailed it down.
Errors is triggered each time an "empty" (with no .md) folder is found.
setting *system.pages.hide_empty_folders: true* was no help

**Comment by gamahachaa** [comment]
Hello Andy is it normal behavior ? We would need to be able to leave empty folders (like media)

**Comment by rhukster** [comment]
this is not normal, sounds like a bug.. i will take a look when i get a moment.. just super busy with client work at th emo.

**Comment by rhukster** [comment]
This seems to be already fixed in TNTSearch 3.0 beta. So i must of found/fixed it at some point.
